contracts/Staking.sol

```diff
@@ -386,11 +384,28 @@ contract Staking is IStaking, Initializable, OwnableUpgradeable, Constants {
         uint256 index = unbondingLength[_source];
         unbondingAmount[_source][index].amount = _amount;
         unbondingAmount[_source][index].startTime = block.timestamp;
+        unbondingAmount[_source][index].indexer = _indexer;
         unbondingLength[_source]++;
 
         emit UnbondRequested(_source, _indexer, _amount, index);
     }
 
+    function cancelUnbonding(uint256 unbondReqId) external {
+        UnbondAmount memory unbond = unbondingAmount[msg.sender][unbondReqId];
+        uint256 time = block.timestamp - unbond.startTime;
+        require(time < lockPeriod, 'unable to cancel');
+        delete unbondingAmount[msg.sender][unbondReqId];
+
+        if (msg.sender != unbond.indexer) {
+            require(
+                delegation[unbond.indexer][unbond.indexer].valueAfter * indexerLeverageLimit >=
+                    totalStakingAmount[unbond.indexer].valueAfter + unbond.amount,
+                'Delegation limitation reached'
+            );
+        }
+        _addDelegation(msg.sender, unbond.indexer, unbond.amount);
```

Comment:
check indexer's stake > 0

Reply:
check current stake amount and next era's stake, because when indexer unregistered in current era, you don't know it if you only check current stake